.github/workflows/build-yocto.yml

```diff
@@ -106,9 +106,21 @@ jobs:
           # Instruct our file server to make these files available for download
           url="https://quic-yocto-fileserver-1029608027416.us-central1.run.app/${GITHUB_RUN_ID}/${{ matrix.machine }}/"
 
-          retries=3
+          retries=4
+          okay=0
+          shopt -s lastpipe  # allows us to capture the value of `okay` in the while loop below
           for ((i=0; i<retries; i++)); do
-              curl -X POST -i -w '%{http_code}' --fail-with-body ${url} && break
+              curl -X POST -H "Accept: text/event-stream" -i --fail-with-body -s -N ${url} | \
+                  while read line; do
+                      echo $line
+                      if [[ $line == STATUS=* ]]; then
+                          if [[ $line == "STATUS=OK" ]]; then
+                              okay=1
+                              break
+                          fi
+                      fi
+                  done
+              [ $okay -eq 1 ] && break
```

Comment:
@doanac I believe that this may not work as expected if the bash script code run with `set -e`. 
In this case when the `okay != 1` the script will abort as the evaluations of the expression will return false, what causes abort when `set -e` is active.

We need to replace that with if conditional statement
```
if [ $okay -eq 1 ]; then
    break
fi
```

Reply:
I didn't test anything to verify what I said but if what I said is true, the consequence is that there are no retries and the loop aborts on the first failure when `okay != 1`.